build-tools/packages/build-cli/src/commands/transform/releaseNotes.ts

```diff
@@ -0,0 +1,109 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { readFile, writeFile } from "node:fs/promises";
+import { Flags } from "@oclif/core";
+import { format as prettier } from "prettier";
+import { remark } from "remark";
+import remarkGfm from "remark-gfm";
+import remarkGithub, { defaultBuildUrl } from "remark-github";
+import admonitions from "remark-github-beta-blockquote-admonitions";
+import remarkToc from "remark-toc";
+
+import { BaseCommand } from "../../library/index.js";
+import {
+	remarkHeadingLinks,
+	removeHeadingsAtLevel,
+	removeSectionContent,
+	stripSoftBreaks,
+	updateTocLinks,
+	// eslint-disable-next-line import/no-internal-modules
+} from "../../library/markdown.js";
+
+/**
+ * Transforms a markdown release notes file into a format appropriate for use in a GitHub Release.
+ */
+export default class TransformReleaseNotesCommand extends BaseCommand<
+	typeof TransformReleaseNotesCommand
+> {
+	static readonly summary =
+		`Transforms a markdown release notes file into a format appropriate for use in a GitHub Release. This is used to transform in-repo release notes such that they can be automatically posted to our GitHub Releases.`;
+
+	// Enables the global JSON flag in oclif.
+	static readonly enableJsonFlag = true;
+
+	static readonly flags = {
+		inFile: Flags.file({
+			description: `A release notes file that was generated using 'flub generate releaseNotes'. The file must not already contain heading links. That is, the input file must not have been generated with the --headingLinks flag.`,
```

Comment:
I'll test it - probably possible to detect but not sure how much it's worth - maybe just stripping anchor tags and regenning them in the remark pipeline will be easier.

Reply:
It turns out that this is the reason to check if the first child node of a heading is text - if I add that check back, then it works even when links have been generated, because it skips them. So I added that check back.